src/clsourcedependencies_parser_test.cc

```diff
@@ -0,0 +1,377 @@
+// Copyright (c) Microsoft Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if NINJA_ENABLE_CL_SOURCE_DEPENDENCIES
+
+#include "clsourcedependencies_parser.h"
+
+#include "test.h"
+
+TEST(ParseCLSourceDependencies, ParseInvalidJSON) {
+  std::string err;
+  std::vector<std::string> includes;
+
+  ASSERT_FALSE(ParseCLSourceDependencies(StringPiece("this is not JSON"), &err,
+                                         includes));
+  ASSERT_EQ("sourceDependencies parse error", err);
+}
+
+TEST(ParseCLSourceDependencies, ParseMissingVersion) {
+  std::string err;
+  std::vector<std::string> includes;
+
+  ASSERT_FALSE(ParseCLSourceDependencies(StringPiece("{}"), &err, includes));
+  ASSERT_EQ("sourceDependencies missing version", err);
+}
+
+TEST(ParseCLSourceDependencies, ParseVersionWrongType) {
+  std::string err;
+  std::vector<std::string> includes;
+
+  ASSERT_FALSE(ParseCLSourceDependencies(StringPiece("{\"Version\":1.0}"), &err,
+                                         includes));
+  ASSERT_EQ("sourceDependencies version must be a string", err);
+}
+
+TEST(ParseCLSourceDependencies, ParseWrongVersion) {
+  std::string err;
+  std::vector<std::string> includes;
+
+  ASSERT_FALSE(ParseCLSourceDependencies(StringPiece("{\"Version\":\"2.0\"}"),
+                                         &err, includes));
+  ASSERT_EQ("expected sourceDependencies version 1.x but found 2.0", err);
+}
+
+TEST(ParseCLSourceDependencies, ParseMissingData) {
+  std::string err;
+  std::vector<std::string> includes;
+
+  ASSERT_FALSE(ParseCLSourceDependencies(StringPiece("{\"Version\":\"1.0\"}"),
+                                         &err, includes));
+  ASSERT_EQ("sourceDependencies missing data", err);
+}
+
+TEST(ParseCLSourceDependencies, ParseDataWrongType) {
+  std::string err;
+  std::vector<std::string> includes;
+
+  ASSERT_FALSE(ParseCLSourceDependencies(
+      StringPiece("{\"Version\":\"1.0\",\"Data\":true}"), &err, includes));
```

Comment:
Consider using C++ literal strings to help readability (in case of mismatch, GoogleTest logs will display both the literal expression from the source code, and the usual expanded version), e.g.: `R"=({"Version": "1.0", "Data": true})="`

This becomes very useful for large hard-coded JSON value.

Reply:
Done.